### PR TITLE
Add a new appxmanifest for preview

### PR DESCRIPTION
**Pull request by carlos-zamora**
## Summary of the Pull Request
This adds a new appxmanifest for 'Windows Terminal (Preview)' and links the resources.

Code-wise, split up `WindowsTerminalReleaseBuild` into...
- WindowsTerminalOfficialBuild: [true, false]
- WindowsTerminalBranding: [Dev, Preview, Release]

Added a comment about that in release.yml

## Validation Steps Performed
used msbuild to build...
- [X] Dev
- [X] Preview
- [X] Release
then checked the msix for the correct name/icon.

### CODE REVIEW

**Comment by carlos-zamora** [comment]
@DHowett-MSFT This is a draft because I'm getting the following error when I try to run CascadiaPackage in VS:
```
30>C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\VisualStudio\v16.0\AppxPackage\Microsoft.AppXPackage.Targets(1194,5): error APPX1501: Project must have either AppxManifest item or Content item with name 'AppxManifest.xml'.
30>Done building project "CascadiaPackage.wapproj" -- FAILED.
```
Ideas?

File: src/cascadia/CascadiaPackage/CascadiaPackage.wapproj [hunk 1]
```diff
@@ -32,10 +32,13 @@
     <None Include="CascadiaPackage_TemporaryKey.pfx" />
   </ItemGroup>
   <ItemGroup>
-    <AppxManifest Include="Package.appxmanifest" Condition="'$(WindowsTerminalReleaseBuild)'=='true'">
+    <AppxManifest Include="Package.appxmanifest" Condition="'$(WindowsTerminalBranding)'=='Release'">
       <SubType>Designer</SubType>
     </AppxManifest>
-    <AppxManifest Include="Package-Dev.appxmanifest" Condition="'$(WindowsTerminalReleaseBuild)'!='true'">
+        <AppxManifest Include="Package-Pre.appxmanifest" Condition="'$(WindowsTerminalBranding)'=='Preview'">
+      <SubType>Designer</SubType>
```
Comment (DHowett-MSFT) [inline src/cascadia/CascadiaPackage/CascadiaPackage.wapproj:39]: make the spacing right here.

File: src/cascadia/CascadiaPackage/CascadiaPackage.wapproj [hunk 2]
```diff
@@ -32,10 +32,13 @@
...
   <ItemGroup>
-    <AppxManifest Include="Package.appxmanifest" Condition="'$(WindowsTerminalReleaseBuild)'=='true'">
+    <AppxManifest Include="Package.appxmanifest" Condition="'$(WindowsTerminalBranding)'=='Release'">
       <SubType>Designer</SubType>
     </AppxManifest>
-    <AppxManifest Include="Package-Dev.appxmanifest" Condition="'$(WindowsTerminalReleaseBuild)'!='true'">
+        <AppxManifest Include="Package-Pre.appxmanifest" Condition="'$(WindowsTerminalBranding)'=='Preview'">
+      <SubType>Designer</SubType>
+    </AppxManifest>
+    <AppxManifest Include="Package-Dev.appxmanifest" Condition="'$(WindowsTerminalBranding)'=='Dev'">
```
Comment (DHowett-MSFT) [inline src/cascadia/CascadiaPackage/CascadiaPackage.wapproj:41]: this is the cause of your VS error. _you MUST have a manifest to use when the variable isn't set_.

For a long time, that was the Dev manifest. We should just keep it that way.

File: src/cascadia/CascadiaPackage/CascadiaPackage.wapproj [hunk 3]
```diff
@@ -46,10 +49,15 @@
     <PRIResource Include="Resources\*\Resources.resw" />
     <PRIResource Include="Resources\Resources.resw" />
   </ItemGroup>
-  <PropertyGroup Condition="'$(WindowsTerminalReleaseBuild)'!='true'">
-    <!-- This is picked up by CascadiaResources.build.items. -->
+
+  <!-- This is picked up by CascadiaResources.build.items. -->
+  <PropertyGroup Condition="'$(WindowsTerminalBranding)'=='Dev'">
```
Comment (DHowett-MSFT) [inline src/cascadia/CascadiaPackage/CascadiaPackage.wapproj:54]: same thing applies here. If you don't set Branding=Dev, and leave it blank, we need to use the dev icons.

File: src/cascadia/CascadiaPackage/Package-Pre.appxmanifest [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
+  xmlns:uap3="http://schemas.microsoft.com/appx/manifest/uap/windows10/3"
+  xmlns:uap4="http://schemas.microsoft.com/appx/manifest/uap/windows10/4"
+  xmlns:uap7="http://schemas.microsoft.com/appx/manifest/uap/windows10/7"
+  xmlns:desktop="http://schemas.microsoft.com/appx/manifest/desktop/windows10"
+  xmlns:rescap="http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities"
+  IgnorableNamespaces="uap mp rescap">
+
+  <Identity
+    Name="Microsoft.WindowsTerminal.Preview"
```
Comment (DHowett-MSFT) [inline src/cascadia/CascadiaPackage/Package-Pre.appxmanifest:15]: Might prefer `WindowsTerminalPreview`

File: src/cascadia/CascadiaPackage/Package-Pre.appxmanifest [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US"
+    Version="1.0.0.0" />
```
Comment (DHowett-MSFT) [inline src/cascadia/CascadiaPackage/Package-Pre.appxmanifest:17]: For fun. call this 0.5.0.0

File: build/pipelines/release.yml [hunk 6]
```diff
@@ -23,17 +31,17 @@ jobs:
   - template: ./templates/build-console-int.yml
     parameters:
       platform: x64
-      additionalBuildArguments: /p:WindowsTerminalReleaseBuild=true
+      additionalBuildArguments: /p:WindowsTerminalOfficialBuild=true;WindowsTerminalBranding=Preview
```
Comment (zadjii-msft) [inline build/pipelines/release.yml:34]: I'm assuming we'll be changing `WindowsTerminalBranding=Preview` to `Release` at some point in the _near_ future?

Comment (DHowett-MSFT) [reply build/pipelines/release.yml:34]: This pipeline isn't actually used, this is just for illustrative purposes :)

Comment (zadjii-msft) [reply build/pipelines/release.yml:34]: oh well then even better 😄

**Pull request merged**